vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/DatabaseBackedProcessor.java

```diff
@@ -778,6 +784,112 @@ record ->
     }
   }
 
+  public final List<String> retry(Optional<List<String>> workflowRunIds) throws SQLException {
+    try (final var connection = dataSource.getConnection()) {
+      final var ids = new ArrayList<String>();
+      DSL.using(connection, SQLDialect.POSTGRES)
+          .transaction(
+              context -> {
+                var dsl = DSL.using(context);
+                var operations =
+                    dsl
+                        .select(ACTIVE_OPERATION.asterisk())
+                        .from(
+                            ACTIVE_OPERATION
+                                .join(ACTIVE_WORKFLOW_RUN)
+                                .on(ACTIVE_OPERATION.WORKFLOW_RUN_ID.eq(ACTIVE_WORKFLOW_RUN.ID))
+                                .join(WORKFLOW_RUN)
+                                .on(WORKFLOW_RUN.ID.eq(ACTIVE_WORKFLOW_RUN.ID)))
+                        .where(
+                            ACTIVE_OPERATION
+                                .STATUS
+                                .eq(OperationStatus.FAILED)
+                                .and(ACTIVE_OPERATION.ENGINE_PHASE.eq(Phase.PROVISION_OUT))
+                                .and(ACTIVE_WORKFLOW_RUN.ENGINE_PHASE.eq(Phase.FAILED))
+                                .and(ACTIVE_OPERATION.ATTEMPT.eq(ACTIVE_WORKFLOW_RUN.ATTEMPT))
+                                .and(
+                                    workflowRunIds
+                                        .map(WORKFLOW_RUN.HASH_ID::in)
+                                        .orElse(trueCondition())))
+                        .stream()
+                        .collect(
+                            Collectors.groupingBy(
+                                r -> r.get(ACTIVE_OPERATION.WORKFLOW_RUN_ID),
+                                Collectors.mapping(
+                                    r ->
+                                        DatabaseOperation.recover(
+                                            r, liveness(r.get(ACTIVE_OPERATION.WORKFLOW_RUN_ID))),
+                                    Collectors.toList())));
+                dsl.update(ACTIVE_WORKFLOW_RUN)
```

Comment:
Requests to the endpoint appear to be hanging at this update()

Reply:
Please provide a list of what you did to set up for your test and how you proceeded to test. Also, have you checked the logs?

Reply:
`mvn clean install` and copied over the jars to vidarr-dev
ran the query `curl -X POST "https://vidarr-dev.gsi.oicr.on.ca/api/retry-provision-out" -H  "accept: application/json" -H  "Content-Type: application/json" -d "{\"workflowRunIds\":[\"exampleid\"]}"`
In the vidarr logs I don't see anything after sending the post, however after restarting the machine I do see: `vidarr-dev.gsi.oicr.on.ca run-vidarr[1062871]: Recovering 12 unstarted workflows fom the database. Retrying workflow exampleid...`
Looking at pg_stat_activity I can see the call has been made but I never get a return:
```
datid            | 30339
datname          | vidarr
pid              | 1062909
leader_pid       | 
usesysid         | 16384
usename          | vidarr
application_name | PostgreSQL JDBC Driver
client_addr      | 127.0.0.1
client_hostname  | 
client_port      | 56238
backend_start    | 2024-02-09 16:03:25.106565+00
xact_start       | 2024-02-09 16:08:14.368482+00
query_start      | 2024-02-09 16:08:14.378947+00
state_change     | 2024-02-09 16:08:14.378948+00
wait_event_type  | Lock
wait_event       | transactionid
state            | active
backend_xid      | 1588026344
backend_xmin     | 1588026343
query            | update "public"."active_workflow_run" set "engine_phase" = $1 where "public"."active_workflow_run"."id" = $2
backend_type     | client backend
```

Reply:
I'm assuming you're substituting `exampleid` for a real ID. What state is `exampleid` in? You will need to have a fresh workflow fail because I changed the way recovery information was stored.

Reply:
`"completed": null,
  "operationStatus": "FAILED",
  "created": "2024-02-07T18:31:43+00:00",`
  I'll retry with a new workflow